src/state/config.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""haproxy-operator charm configuration."""
+
+import itertools
+import logging
+import typing
+
+import ops
+from pydantic import Field, ValidationError
+from pydantic.dataclasses import dataclass
+
+logger = logging.getLogger()
+
+
+class InvalidCharmConfigError(Exception):
+    """Exception raised when a charm configuration is found to be invalid."""
+
+
+@dataclass(frozen=True)
+class CharmConfig:
+    """A component of charm state that contains the charm's configuration.
+
+    Attributes:
+        global_max_connection: The configured gateway class.
+    """
+
+    global_max_connection: int = Field(gt=0)
+
+    @classmethod
+    def from_charm(cls, charm: ops.CharmBase) -> "CharmConfig":
+        """Create a CharmConfig class from a charm instance.
+
+        Args:
+            charm: The gateway-api-integrator charm.
+
+        Raises:
+            InvalidCharmConfigError: When the chamr's config is invalid.
```

Comment:
s/chamr/charm/

Reply:
Thanks, this was fixed in future PRs but I haven't back ported it to this one. I just fixed this typo.